res/controllers/common-controller-scripts.js

```diff
@@ -28,6 +28,44 @@ String.prototype.toInt = function() {
     return a;
 };
 
+/**
+ * Discriminates whether an object was created using the `{}` synthax.
+ *
+ * Returns true when was an object was created using the `{}` synthax.
+ * False if the object is an instance of a class like Date or Proxy or an Array.
+ *
+ * Object.isSimpleObject({}) // true
+ * Object.isSimpleObject(null) // false
+ * Object.isSimpleObject(undefined) // false
+ * Object.isSimpleObject(new Date) // false
+ * Object.isSimpleObject(new (class {})()) // false
+ * @param {any} obj Object to test
+ * @returns {boolean} true if obj was created using the `{}` synthax, false otherwise
+ */
+Object.isSimpleObject = function(obj) {
+    return obj !== null && typeof obj === "object" && obj.constructor.name === "Object";
+};
+
+/**
+ * Deeply merges 2 objects.
+ * @param source {object | Array} Object to merge target into
+ * @param target {object | Array} Object to merge into target
+ */
+Object.deepMerge = function(source, target) {
+    if (source === target) {
+        return;
+    }
+
+    Object.keys(source).forEach(k => {
+        if (Array.isArray(source[k]) && Array.isArray(target[k])) {
+            source[k].push(...target[k]);
+        } else if (Object.isSimpleObject(source[k]) && Object.isSimpleObject(target[k])) {
+            Object.deepMerge(source[k], target[k]);
+        }
+        Object.assign(source, target);
```

Comment:
I'm not sure how critical it is, but `_.merge({a:1},{a:undefined})` and `Object.deepMerge({a:1},{a:undefined})` will differ in behavior due to the fact that `Object.assign` will overwrite with `undefined` and `_.merge` will not.

Reply:
Right. This is fixed now.